protocol/contracts/core/MultiOtcOffer.sol

```diff
@@ -0,0 +1,187 @@
+pragma solidity 0.8.19;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (core/MultiOtcOffer.sol)
+
+import { Pausable } from "@openzeppelin/contracts/security/Pausable.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { ElevatedAccess } from "../nexus/access/ElevatedAccess.sol";
+import { IMultiOtcOffer } from "../interfaces/core/IMultiOtcOffer.sol";
+
+
+/**
+ * @title Multi OTC Offer
+ *
+ * @notice Temple offers OTC purchases to users on certain tokens - slippage and price impact free.
+ * Temple sets the offer price and users can swap tokens for any arbitrary size at this price, up to some
+ * max amount of treasury funds (determined by the `fundsOwner` balance and ERC20 approvals). 
+ * This contract is set up to take care of multiple OTC markets. Contract owner can add and remove OTC markets.
+ */
+contract MultiOtcOffer is IMultiOtcOffer, Pausable, ElevatedAccess {
+    using SafeERC20 for IERC20Metadata;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    /// @notice The number of decimal places represented by `offerPrice`
+    uint8 public constant OFFER_PRICE_DECIMALS = 18;
+
+    mapping(bytes32 => OTCMarketInfo) public otcMarketInfo;
+    mapping(bytes32 => OTCMarketPriceParams) public otcMarketPriceParams;
```

Comment:
I think it would be simpler if there was just one struct which contained all the market and pricing info? Having two separate mappings makes it a bit more complicated when you need all the info anyway in a swap anyway?

Reply:
I was considering the setters in this decision. 2 setters `setOfferPriceRange` and `setOfferPrice` access the price params and the other access just the info. 
But I think swap is used more often so will merge the struct